### PR TITLE
Pull all accounts, Find best match when more than one contact returne…

### PR DESCRIPTION
**Pull request by EvaPaunovaPhwk**
…d, Round expense amount

### CODE REVIEW

File: src/services/xero/client/accounting/Client.ts [hunk 1]
```diff
@@ -72,40 +72,27 @@ export class Client implements IClient {
...
-        ];
-        const expenseAccounts = ObjectSerializer.deserialize<IAccountCode[]>(responseItems);
-        return expenseAccounts;
-    }
-
-    async loadExpenseAccount(whereFilters: string[], status?: AccountStatus) {
+        let whereFilter;
         if (status) {
-            whereFilters.push(`Status=="${status}"`);
+            whereFilter = `whereStatus=="${status}"`;
```
Comment (vassildinev) [inline src/services/xero/client/accounting/Client.ts:77]:
```suggestion
            whereFilter = `Status=="${status}"`;
```

File: src/services/payhawk/Client.ts [hunk 2]
```diff
@@ -29,6 +29,11 @@ export class Client implements IClient {
                     json: true,
                     headers: this.headers,
                 });
+            // temporary fix, the problem is that the expense amount is not rounded because it's amount is calculated with myriadsToNumber
+            // payment on the other hand is rounded (myriadsToAmount)
+            // this causes expense with amount 6.81 to have reconciliation.totalAmount = 6.8100000000000005 and payment amount = 6.81
+            // and because of this we hit "Payment total amount does not cover expense total amount"
+            expenseResult.reconciliation.totalAmount = expenseResult.reconciliation.totalAmount.toFixed(2);
```
Comment (vassildinev) [inline src/services/payhawk/Client.ts:36]: isn't the fix then more correct to be that total amount is also calculated using myriadthsToAmount ?

File: src/services/xero/client/accounting/Client.ts (hunk 1 above)
Comment (EvaPaunovaPhwk) [reply src/services/xero/client/accounting/Client.ts]: Fixed

File: src/services/payhawk/Client.ts (hunk 2 above)
Comment (EvaPaunovaPhwk) [reply src/services/payhawk/Client.ts:36]: Discussed offline

**Review by vassildinev — APPROVED** [review]

**Pull request merged**